File: 006/Haskell/Problem006.hs
module Problem006 (diff_squares_sums, sum_of_squares, square_of_sum) where

diff_squares_sums :: Integer -> Integer
diff_squares_sums n = (square_of_sum n) - (sum_of_squares n)

sum_of_squares n = n * (n + 1) * (2 * n + 1) `div` 6

square_of_sum n = (n * (n + 1) `div` 2)^2

File: 006/Ruby/problem006.rb
class Integer

  def diff_squares_sums
    square_of_sum - sum_of_squares
  end

  def sum_of_squares
    self * (self + 1) * (2 * self + 1) / 6
  end

  def square_of_sum
    sum = self * (self + 1) / 2
    sum * sum
  end

end
